lib/globus/client/endpoint.rb

```diff
@@ -98,29 +83,58 @@ def paths
         end
       end
 
+      # @see #paths
+      def full_path
+        paths.last
+      end
+
       def path_segments
         [user_id, "work#{work_id}", "version#{work_version}"]
       end
 
+      def access_request(permissions:)
+        response = connection.post(access_path) do |req|
+          req.body = {
+            DATA_TYPE: "access",
+            principal_type: "identity",
+            principal: user,
+            path: full_path,
+            permissions:,
+            notify_email: "#{user_id}@stanford.edu"
+          }.to_json
+          req.headers["Content-Type"] = "application/json"
+        end
+
+        return response if response.success?
+
+        # Ignore error if permissions already set for identity
+        if response.status == 409
+          error = JSON.parse(response.body)
+          return if error["code"] == "Exists"
+        end
+
+        UnexpectedResponse.call(response)
+      end
+
       def objects
         # List files at an endpoint https://docs.globus.org/api/transfer/file_operations/#list_directory_contents
-        response = connection.get("#{transfer_path}/ls?path=#{paths.last}")
+        response = connection.get("#{transfer_path}/ls?path=#{full_path}")
         return JSON.parse(response.body) if response.success?
 
         UnexpectedResponse.call(response)
       end
 
+      def files
+        objects["DATA"].select { |object| object["DATA_TYPE"] == "file" }
+      end
```

Comment:
Moved this up so `files` and `objects` could be buddies, and the `_path` methods below could be buddies.